src/main/scala/firrtl/passes/LowerTypes.scala

```diff
@@ -296,11 +304,11 @@ private object DestructTypes {
     *         instead of a flat Reference when turning them into access expressions.
     */
   def destructInstance(
-    m:         ModuleTarget,
-    instance:  DefInstance,
-    namespace: Namespace,
-    renameMap: RenameMap,
-    reserved:  Set[String]
+    m:           ModuleTarget,
+    instance:    DefInstance,
+    namespace:   Namespace,
+    instRenames: mutable.ListBuffer[(Instance, Instance)],
```

Comment:
since you are appending, should this better be an ArrayBuffer?

Reply:
`ListBuffer` has constant time for both prepend and append: https://www.scala-lang.org/api/2.12.11/scala/collection/mutable/ListBuffer.html. I use `ListBuffer` whenever I want to mutably build up an immutable `Seq` because `ListBuffer.toList` is free (copy on write to the underlying buffer).

Reply:
Cool. I learned something!

Reply:
Yeah it's pretty neat. I think once we're on 2.13 I might shift to using `VectorBuilder` since `Vector` is much better in `2.13.2` and has better memory characteristics than `List`.